index.js

```diff
@@ -26,26 +25,21 @@ module.exports = function () {
 		}
 
 		return fullname;
-	}).catch(fallback).catch(function () {});
+	}).catch(fallback).catch(() => {});
 };
 
 function fallback() {
 	if (process.platform === 'darwin') {
-		return exec('id -P')
-			.then(function (stdout) {
-				fullname = stdout.trim().split(':')[7];
-
-				if (!fullname) {
-					throw new Error();
-				}
+		return passwdUser(process.getuid())
+			.then(user => {
+				fullname = user.fullname;
 
 				return fullname;
```

Comment:
I was thinking of doing `.then(user => fullname = user.fullname)` (for all fallbacks) and then `return fallback().then(() => username);` on line 24 instead to make it shorted, but opted not to since this is more clear. Tell me if you like it the other way.